spring-cloud-aws-autoconfigure/src/main/java/org/springframework/cloud/aws/autoconfigure/context/ContextResourceLoaderAutoConfiguration.java

```diff
@@ -29,6 +30,7 @@
  * @author Agim Emruli
  */
 @Configuration
+@ConditionalOnAwsCloudEnvironment
```

Comment:
I have experimented here with a new `@ConditionalOnAwsRegionAvailable` implementation that tries the available `RegionProvider`; if a static region is set the condition can pass outside EC2.